libraries/AP_Baro/AP_Baro_ICM20789.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "AP_Baro_Backend.h"
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/Semaphores.h>
+#include <AP_HAL/Device.h>
+
+#ifndef HAL_BARO_ICM20789_I2C_ADDR
+// the baro is on a separate I2C address from the IMU, even though in
+// the same package
+#define HAL_BARO_ICM20789_I2C_ADDR 0x63
+#endif
+
+class AP_Baro_ICM20789 : public AP_Baro_Backend
+{
+public:
+    void update();
+
+    static AP_Baro_Backend *probe(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::Device> dev);
+    
+private:
+    AP_Baro_ICM20789(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::Device> dev);
+    virtual ~AP_Baro_ICM20789(void) {};
+    
+    bool init();
+    bool send_cmd16(uint16_t cmd);
+
+    bool read_calibration_data(void);
+
+    void convert_data(uint32_t Praw, uint32_t Traw);
+
+    void calculate_conversion_constants(const float p_Pa[3], const float p_LUT[3],
+                                        float &A, float &B, float &C);
+    float get_pressure(uint32_t p_LSB, uint32_t T_LSB);
+
+    bool spi_init(void);
+    
+    void timer(void);
+    
+    // calibration data
+    int16_t sensor_constants[4];
+
+    uint8_t instance;
+    AP_HAL::OwnPtr<AP_HAL::Device> dev;
+    AP_HAL::OwnPtr<AP_HAL::Device> dev_icm;
```

Comment:
Why keep *dev_icm*? Looks like it is only needed for initialization.

Reply:
Will make it a stack variable.